### PR TITLE
Get TIF Source Config API

### PR DESCRIPTION
**Pull request by jowg-amazon**
### Description
Get TIF Source Config API by ID
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigAction.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+package org.opensearch.securityanalytics.threatIntel.action;
+
+import org.opensearch.action.ActionType;
+
+import static org.opensearch.securityanalytics.threatIntel.sacommons.IndexTIFSourceConfigAction.GET_TIF_SOURCE_CONFIG_ACTION_NAME;
+
+/**
+ * Threat intel tif job creation action
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigAction.java:13]: correct java doc

File: src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigRequest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Get threat intel feed source config request
+ */
+public class SAGetTIFSourceConfigRequest extends ActionRequest {
+    private final String id;
+    private final Long version;
+    public static final String TIF_SOURCE_CONFIG_ID = "tif_id";
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigRequest.java:24]: tif_source_config_id?

File: src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigResponse.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.securityanalytics.threatIntel.model.SATIFSourceConfigDto;
+
+import java.io.IOException;
+
+import static org.opensearch.securityanalytics.util.RestHandlerUtils._ID;
+import static org.opensearch.securityanalytics.util.RestHandlerUtils._VERSION;
+
+public class SAGetTIFSourceConfigResponse extends ActionResponse implements ToXContentObject {
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigResponse.java:20]: add response serde tests

File: src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigRequest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.io.IOException;
+import java.util.Locale;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Get threat intel feed source config request
+ */
+public class SAGetTIFSourceConfigRequest extends ActionRequest {
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigRequest.java:21]: add request serde tests 

plz add tests even for Create config req/res

File: src/main/java/org/opensearch/securityanalytics/threatIntel/dao/SATIFSourceConfigDao.java [hunk 5]
```diff
@@ -148,21 +156,60 @@ public void createJobIndexIfNotExists(final StepListener<Void> stepListener) {
...
+                        XContentParser xcp = XContentHelper.createParser(
+                                xContentRegistry, LoggingDeprecationHandler.INSTANCE,
+                                response.getSourceAsBytesRef(), XContentType.JSON
+                        );
+                        SaTifSourceConfig = SATIFSourceConfig.docParse(xcp, response.getId(), response.getVersion());
+                        assert SaTifSourceConfig != null;
+                    }
+                    log.debug("Threat intel source config with id [{}] fetched.", response.getId());
+                    actionListener.onResponse(SaTifSourceConfig);
+                } catch (IOException ex) {
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/dao/SATIFSourceConfigDao.java:203]: log

File: src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigAction.java (hunk 1 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigAction.java]: fixed

File: src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigRequest.java (hunk 2 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigRequest.java]: changed to `tif_source_config_id`

File: src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigResponse.java (hunk 3 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/threatIntel/action/SAGetTIFSourceConfigResponse.java:20]: added tests

File: src/main/java/org/opensearch/securityanalytics/threatIntel/dao/SATIFSourceConfigDao.java (hunk 5 above)
Comment (jowg-amazon) [reply src/main/java/org/opensearch/securityanalytics/threatIntel/dao/SATIFSourceConfigDao.java:203]: added error log

**Review by eirsep — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**